src/riemannian_diff.jl

```diff
@@ -0,0 +1,232 @@
+[`rdifferential_backend!`](@ref)
+"""
+mutable struct CurrentRiemannianDiffBackend
+    backend::AbstractRiemannianDiffBackend
+end
+
+"""
+    _current_rgradient_backend
+
+The instance of [`Manifolds.CurrentRiemannianDiffBackend`](@ref) that stores the
+globally default differentiation backend for calculating gradients.
+
+# See also
+
+[`Manifolds.gradient(::Any, ::RealField, ::Any, ::AbstractRiemannianDiffBackend)`](@ref)
+"""
+const _current_rgradient_backend = CurrentRiemannianDiffBackend(RiemannianONBDiffBackend(
+    diff_backend(),
+    ExponentialRetraction(),
+    LogarithmicInverseRetraction(),
+    DefaultOrthonormalBasis(),
+),)
+
+"""
+    _current_rdifferential_backend
+
+The instance of [`Manifolds.CurrentRiemannianDiffBackend`](@ref) that stores the
+globally default differentiation backend for calculating differentials.
+
+# See also
+
+[`Manifolds.differential`](@ref)
+"""
+const _current_rdifferential_backend =
+    CurrentRiemannianDiffBackend(RiemannianONBDiffBackend(
+        diff_backend(),
+        ExponentialRetraction(),
+        LogarithmicInverseRetraction(),
+        DefaultOrthonormalBasis(),
+    ),)
+
+"""
+    rgradient_backend() -> AbstractRiemannianDiffBackend
+
+Get the current differentiation backend for Riemannian gradients.
+"""
+rgradient_backend() = _current_rgradient_backend.backend
+
+"""
+    rgradient_backend!(backend::AbstractRiemannianDiffBackend)
+
+Set current Riemannian gradient backend for differentiation to `backend`.
+"""
+function rgradient_backend!(backend::AbstractRiemannianDiffBackend)
+    _current_rgradient_backend.backend = backend
+    return backend
+end
+
+"""
+    rdifferential_backend() -> AbstractRiemannianDiffBackend
+
+Get the current differentiation backend for Riemannian differentials.
+"""
+rdifferential_backend() = _current_rdifferential_backend.backend
+
+"""
+    rdifferential_backend!(backend::AbstractRiemannianDiffBackend)
+
+Set current Riemannian differential backend for differentiation to `backend`.
+"""
+function rdifferential_backend!(backend::AbstractRiemannianDiffBackend)
+    _current_rdifferential_backend.backend = backend
+    return backend
+end
+
+"""
+    RiemannianProjectionGradientBackend(
+        diff_backend::AbstractDiffBackend
+    ) <: AbstractRiemannianDiffBackend
+
+Riemannian differentiation based on differentiation in the ambient space and projection to
+the given manifold. Differentiation in the ambient space is performed using
+the backend `diff_backend`.
+
+Only valid for manifolds that are embedded in a special way in the Euclidean space.
+See [^Absil2008], Section 3.6.1 for details.
+
+[^Absil2008]:
+    > Absil, P. A., et al. Optimization Algorithms on Matrix Manifolds. 2008.
+"""
+struct RiemannianProjectionGradientBackend{TADBackend<:AbstractDiffBackend} <:
+       AbstractRiemannianDiffBackend
+    diff_backend::TADBackend
+end
+
+function gradient(f, ft::RealField, p, backend::RiemannianProjectionGradientBackend)
```

Comment:
I wanted to ask about this approach already in the last variant, but yes, this is quite nice and general.